event/system.go

```diff
@@ -0,0 +1,153 @@
+package event
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Subscriber is a limited interface of System that only allows subscribing and
+// unsubscribing.
+type Subscriber interface {
+	Subscribe(events ...Type) (subID uint64, eventsCh <-chan *Event)
+	Unsubscribe(subID uint64)
+}
+
+// System keeps track of subscribers, and allows subscribing to and emitting
+// events.
+type System struct {
+	ctx         context.Context
```

Comment:
Yeah, not even `Emit()`, since it's only used in the function returned from `Emit()`, so we can pass it there. I wanted to keep the signatures simple, but sure, I'll remove it from the struct.

Reply:
I removed it in 8658f68b9.

Since it doesn't make sense to pass both a `time.Duration` and a context to the wait function, I also removed the `time.Duration`, as we can pass a context that times out, which serves the same purpose.